src/ir_visitor/visitor.rs

```diff
@@ -96,11 +49,13 @@ impl<T: Default> Construct for T {
     }
 }
 
-/// A visitor for the commands by detaching them from the underlying component.
-pub trait Visitor
+/// A Visitor for the commands by detaching them from the underlying component.
+pub trait Visitor<'comp>
 where
     Self: Sized + Construct,
 {
+    type Data: VData<'comp>;
```

Comment:
Why does the data need the lifetime parameter to come from the trait instead of the struct that implements the trait? It seems we don't use this parameter except to pass to the VData field.

Reply:
I think you probably want an `ExtraData` field that stores other information that is beyond what `VisitorData` currently stores and expose that as a `self.get_data` and `self.get_mut_data` methods or something.

Reply:
It doesn't compile if you don't have `'comp` because the `new` function needs to generate a `Self` that lasts long enough. 
![image](https://github.com/cucapra/filament/assets/34429084/58c4b1f3-4771-4bbf-929f-e5d2e37f2bca)
In the implementation of `new` I need `'comp` in the signature which also requires me to define it in `VData`.

Reply:
Error if `VData` doesn't define `'comp`: 
![image](https://github.com/cucapra/filament/assets/34429084/256179f6-1b51-43bf-b920-2dfd46ab5cb9)

Reply:
I also can't just do the following because adding `'comp` changes the signature of the `new` function...
![image](https://github.com/cucapra/filament/assets/34429084/5f9c8473-0c08-4513-9f55-763e7e6546e0)